packages/ingest/granule.js

```diff
@@ -764,43 +741,57 @@ async function moveGranuleFile(source, target, options) {
+ * @param {Array<Object>} sourceFiles - array of file objects, they are updated with destination
+ * location after the files are moved
+ * @param {string} sourceFiles.name - file name
+ * @param {string} sourceFiles.bucket - current bucket of file
+ * @param {string} sourceFiles.filepath - current s3 key of file
  * @param {Object[]} destinations - array of objects defining the destination of granule files
- * @param {string} destinations[].regex - regex for matching filepath of file to new destination
- * @param {string} destinations[].bucket - aws bucket
- * @param {string} destinations[].key - filepath on the bucket for the destination
+ * @param {string} destinations.regex - regex for matching filepath of file to new destination
+ * @param {string} destinations.bucket - aws bucket of the destination
+ * @param {string} destinations.filepath - file path/directory on the bucket for the destination
  * @param {string} distEndpoint - distribution enpoint from config
- * @returns {Promise<undefined>} returns `undefined` when all the files are moved
+ * @param {boolean} published - indicate if publish is needed
+ * @returns {Promise<Object>} returns promise from publishing cmr file or {}
  */
-async function moveGranuleFiles(sourceFiles, destinations, distEndpoint) {
+async function moveGranuleFiles(granuleId, sourceFiles, destinations, distEndpoint, published) {
   const moveFileRequests = sourceFiles.map((file) => {
     const destination = destinations.find((dest) => file.name.match(dest.regex));
-
-    if (file.name.match(/.*\.cmr\.xml$/)) {
-      return updateMetadata(
-        { file, destination }, sourceFiles, destinations, distEndpoint
-      );
-    }
+    const parsed = aws.parseS3Uri(file.filename);
     // if there's no match, we skip the file
-    else if (destination) {
+    if (destination) {
       const source = {
-        Bucket: file.bucket,
-        Key: file.filepath
+        Bucket: parsed.Bucket,
+        Key: parsed.Key
       };
 
       const target = {
         Bucket: destination.bucket,
-        Key: `${destination.filepath}/${file.name}`
+        Key: urljoin(destination.filepath, file.name)
       };
 
-      return moveGranuleFile(source, target);
+      log.debug('moveGranuleFiles', source, target);
+      return moveGranuleFile(source, target).then(() => { /* eslint-disable no-param-reassign */
+        // update the granule file location in source file
+        file.bucket = target.Bucket;
+        file.filepath = target.Key;
+        file.filename = aws.buildS3Uri(file.bucket, file.filepath);
+      });
     }
+    // else set filepath as well so it won't be null
+    file.filepath = parsed.Key;
+    return Promise.resolve();
   });
 
-  return Promise.all(moveFileRequests);
+  await Promise.all(moveFileRequests);
+
+  // update cmr metadata with new file urls
+  const xmlFile = sourceFiles.filter((file) => file.name.endsWith('cmr.xml'));
+  if (xmlFile.length === 1) {
```

Comment:
Should we do > 1 here just to be safe?

Reply:
I think a granule should not have more then one cmr.xml file.  If it does, are we going to update them all?

Reply:
Yeah you're probably right

Reply:
Yeah good call. It would be good to at least log it.